credentials.go

```diff
@@ -145,24 +147,73 @@ func ParseKeyFile(location string) ([]byte, error) {
 	return ParseKeyData(data)
 }
 
-var keyDataPattern = regexp.MustCompile(`<Data>(.+)</Data>`)
+type xmlKeyFileData struct {
+	XMLName xml.Name       `xml:"KeyFile"`
+	Meta    xmlKeyFileMeta `xml:"Meta"`
+	Key     xmlKeyFileKey  `xml:"Key"`
+}
+
+type xmlKeyFileMeta struct {
+	Version string `xml:"Version"`
+}
+
+type xmlKeyFileKey struct {
+	Hash string `xml:"Hash,attr"`
+	Data []byte `xml:"Data"`
+}
+
+var whiteSpacePattern = regexp.MustCompile(`\s+`)
+
+const xmlKeyDataHashLength = 4
+
+// var keyDataPattern = regexp.MustCompile(`<Data>(.+)</Data>`)
 
 // ParseKeyData returns the hashed key from a key file in bytes, parsing xml if needed
 func ParseKeyData(data []byte) ([]byte, error) {
-	if keyDataPattern.Match(data) { //If keyfile is in xml form, extract key data
-		base := keyDataPattern.FindSubmatch(data)[1]
-		data = make([]byte, base64.StdEncoding.DecodedLen(len(base)))
-		if _, err := base64.StdEncoding.Decode(data, base); err != nil {
-			return nil, err
+	keyFileData := xmlKeyFileData{}
+	err := xml.Unmarshal(data, &keyFileData)
+	if err == nil {
+		keyFileData.Key.Data = whiteSpacePattern.ReplaceAll(keyFileData.Key.Data, []byte(``))
+
+		switch keyFileData.Meta.Version {
+		case "1.00", "1.0":
+			decodedKey := make([]byte, base64.StdEncoding.DecodedLen(len(keyFileData.Key.Data)))
+			_, err := base64.StdEncoding.Decode(decodedKey, keyFileData.Key.Data)
+			return decodedKey, err
+		case "2.0":
```

Comment:
Can it be also declared as "2.00"?

Reply:
According to the comment in the original source it was only `1.00` for some versions of the code, and the other would be `1.0` and `2.0`